etcdmain/config.go

```diff
@@ -212,6 +212,9 @@ func (cfg *config) Parse(arguments []string) error {
 	default:
 		os.Exit(2)
 	}
+	if len(cfg.FlagSet.Args()) != 0 {
+		return fmt.Errorf("unsupported non-flag arguments %v", cfg.FlagSet.Args())
```

Comment:
The remaining part is not flag because they don't have prefix '-' or '--'.
what about `%v are not valid etcd arguments`?
and having punctuation in error messages is not good IMO